compositor_pipeline/src/audio_mixer/prepare_inputs/consecutive_frames_tests.rs

```diff
@@ -0,0 +1,246 @@
+                    start_pts: first_batch_start,
+                    end_pts: first_batch_start + (4 * sample_duration)
+                },
+                InputSamples {
+                    samples: second_batch.clone(),
+                    start_pts: second_batch_start,
+                    end_pts: second_batch_start + (4 * sample_duration)
+                },
+            ],
+            sample_rate
+        ),
+        vec![(2, 2), (3, 3), (4, 4), (5, 5), (6, 6), (7, 7)]
+    );
+    assert_eq!(
+        frame_input_samples(
+            start + batch_duration,
+            end + batch_duration,
+            vec![
+                InputSamples {
+                    samples: second_batch.clone(),
+                    start_pts: second_batch_start,
+                    end_pts: second_batch_start + (4 * sample_duration)
+                },
+                InputSamples {
+                    samples: third_batch.clone(),
+                    start_pts: third_batch_start,
+                    end_pts: third_batch_start + (4 * sample_duration)
+                }
+            ],
+            sample_rate
+        ),
+        vec![(8, 8), (9, 9), (10, 10), (11, 11), (12, 12), (0, 0)]
+    );
+
+    // shifted by small_error (add)
+    let first_batch_start = start - sample_duration + small_error;
+    let second_batch_start = first_batch_start + (4 * sample_duration);
+    let third_batch_start = first_batch_start + (8 * sample_duration);
+    assert_eq!(
+        frame_input_samples(
+            start,
+            end,
+            vec![
+                InputSamples {
+                    samples: first_batch.clone(),
+                    start_pts: first_batch_start,
+                    end_pts: first_batch_start + (4 * sample_duration)
+                },
+                InputSamples {
+                    samples: second_batch.clone(),
+                    start_pts: second_batch_start,
+                    end_pts: second_batch_start + (4 * sample_duration)
+                },
+            ],
+            sample_rate
+        ),
+        vec![(1, 1), (2, 2), (3, 3), (4, 4), (5, 5), (6, 6)]
+    );
+    assert_eq!(
+        frame_input_samples(
+            start + batch_duration,
+            end + batch_duration,
+            vec![
+                InputSamples {
+                    samples: second_batch.clone(),
+                    start_pts: second_batch_start,
+                    end_pts: second_batch_start + (4 * sample_duration)
+                },
+                InputSamples {
+                    samples: third_batch.clone(),
+                    start_pts: third_batch_start,
+                    end_pts: third_batch_start + (4 * sample_duration)
+                }
+            ],
+            sample_rate
+        ),
+        vec![(7, 7), (8, 8), (9, 9), (10, 10), (11, 11), (12, 12)]
+    );
+
+    // shifted by small_error (subtract) + batches overlapping between frames
```

Comment:
Where do those overlap? I don't understand this

Reply:
third_batch_start is slightly smaller than second_batch_start, but output is still continous